test/golden_output_test.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Artman golden tests.
+
+It runs GAPIC generation against the golden input data, can then compare the
+output layout with the expected output. This helps to capture unexpected GAPIC
+output layout change caused by either by artman, toolkit or other runtime
+dependency.
+
+Note: a set of static input files (e.g. google/api/http.proto) are being used
+for the golden test, so it is not going to capture upstream changes to such
+input.
+"""
+
+import os
+import subprocess
+
+
+def run_golden_test():
```

Comment:
I just looked at the code, and we are using `py.test`.

So:

  1. I would ask that you make the module conform to py.test idioms (whether or not you conform to `unittest.TestCase` idioms -- either yes or no is fine with me). In particular that means `test_*.py` files and functions named according to `test_*`. We can use nox to separate unit from system from golden (??) tests.
  2. Similarly `py.test` has _excellent_ stack traces and output and test failure messages, even if you just use `assert` (the keyword). Move to that.
  3. _Do_ separate this from actual unit tests (as you have done).